PR3: anchor() page (#8)
chrisdavidmills/content: files/en-us/web/css/anchor/index.md

{{CSSRef}}

> The **`anchor()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) is used within an **anchor-positioned** element's [inset property](#properties_that_accept_anchor_function_values) value, returning a length value relative to the position of the edges of its associated **anchor element**.

Author: it isn't necessarily the value; but part of the value

The `anchor()` function's syntax is as follows:

```text

> anchor(<anchor-element> <anchor-side>, <length-percentage>)

Author: only keywords don't get the brackets. Please update all the syntaxes with `<` and `>`

Reviewer: OK, I've checked all the function pages and corrected these.

    - `left`
      - : The left of the anchor element
    - `start`
      - : The logical start of the anchor element's [block axis](/en-US/docs/Glossary/Logical_properties#block_direction).
    - `end`
      - : The logical end of the anchor element's block axis.
    - `self-start`
      - : The logical start of the anchor element's [inline axis](/en-US/docs/Glossary/Logical_properties#inline_direction).
    - `self-end`
      - : The logical end of the anchor element's inline axis.
    - `center`
      - : The center of the axis of the inset property on which the `anchor()` function is set.
    - {{cssxref("percentage")}}
      - : Specifies the distance, as a percentage, from the start of the axis of the inset property on which the `anchor()` function is set.

> The CSS anchor positioning module introduces two addition `<anchor-side>` values, `inside` and `outside`, that have not yet been implemented.

Author: copied the sentence structure from a different page that has unimplemented values from a datatype.

## Description

> The `anchor()` function enables positioning an element relative to the edges of an anchor element. It is only valid within withing the value of {{glossary("inset properties")}}.

Author: ```suggestion
The `anchor()` function enables positioning an element relative to the edges of an anchor element. It is only valid within the value of {{glossary("inset properties")}}.
```

The `anchor()` function resolves to a `<length>`, so it can be used within [other CSS functions](/en-US/docs/Web/CSS/CSS_Functions) that accept length values, including {{cssxref("calc()")}}, {{cssxref("clamp()")}}, etc.

The function sets the inset position value as an absolute distance relative to the anchor element by defining the anchor element, the side of the anchor element the positioned element is being positioned relative to, and the distance from that side.

> If a positioned element's {{cssxref("position-anchor")}} property value differs from the target anchor listed as the `<anchor-element>` the target listed within the `anchor()` function takes precedence. If the target anchors listed in the inline-direction inset property anchor function differs from the one listed in the block direction, the `<anchor-element>` listed within the `anchor()` block direction's inset property takes precedence.

Author: i did a poop ton of testing, and these three paragraphs are the results of that. We likely should confirm that my findings are accurate.

If the positioned element does not have an anchor associated with it (i.e. via the {{cssxref("position-anchor")}} property) and it has inset properties containing `anchor()` functions within the value, the fallback `<lenght-perecentage>` value is used if one is available. For example, if `top: anchor(bottom, 50px)` were specified on the positioned element but no anchor was associated with it, the fallback value would be used, so `top` would get a computed value of `50px`. If no fallback is available, the inset properties have no effect, but is still valid (it is not considered [invalid CSS](/en-US/docs/Web/CSS/CSS_syntax/Error_handling) and is therefore not ignored).

> For {{glossary("physical properties", "physical")}} inset values, the anchor side you position the positioned element relative to has to be along the same axis as the inset value being set. For example, `top: anchor(left)` is invalid. If `top: anchor(left, 50px)` were specified, the fallback value would be used, so `top` would get a computed value of `50px`. If no fallback is present, again, the inset property has no effect, as if the inset property were set to `auto`.

Author: i added an invalid value in the example

When the `anchor()` function refers to a side of the default anchor, you can include a {{cssxref("margin")}} to create spacing between the edges of the anchor and positioned element as needed. Alternatively, include the `anchor()` function within a {{cssxref("calc")}} function to add spacing.

> This example position the right edge of the positioned element flush to the anchor element's left edge then add margin to make some space between the adges \*/:

Author: ```suggestion
This example positions the right edge of the positioned element flush to the anchor element's left edge then adds a  {{cssxref("margin")}} to create space between the edges:
```

This example position the right edge of the positioned element flush to the anchor element's left edge then add margin to make some space between the adges \*/:

```css
.positionedElement {

>   /* Position the right edge of the positioned element

Author: please do this to all comments in all code blocks on all the pages.

Reviewer: I've checked all the PRs to see if there are any other comments that could do with converting to text; there isn't, so we are all good here.

form div:last-child {
  margin-top: 10px;
}
```

```css
.anchor {
  anchor-name: --myAnchor;
}
```

The positioned element is then associated with the anchor element by setting its anchor name as the value of the positioned element's `position-anchor` property.

In addition, we set the `infobox`'s:

> - {{cssxref("position")}} property to `fixed`, so it can be positioned relative to the anchor's position in the page.

Author: we don't need to include an explanation for every property in the spec; just those that would be new to the reader of the current page. the person using the anchor function likely knows "position" and "anchor-name".  We can cover all the css in a paragraph and then just have a single code block.

Please do this to all "advanced" anchor position property pages.

Reviewer: I've checked through all the PRs, and made a few fixes.

```

```css
.infobox {

>   position-anchor: --myAnchor;

Author: this page shows how a way to include the anchor name. let's include what is covered in the page in the example!

}

.infobox {
  position: fixed;

>   top: anchor(--myAnchor bottom);

Author: :D

topSelect.addEventListener("change", (e) => {
  const anchorSide = e.target.value;

>   infobox.style.top = `anchor(--myAnchor ${anchorSide})`;

Author: Did not test to see if overriding top would override the anchor name in top; but simple enough to include it in our js.

## See also

- {{cssxref("position-anchor")}}
- {{cssxref("inset-area")}}
- [Using CSS anchor positioning](/en-US/docs/Web/CSS/CSS_anchor_positioning/Using)
- [Handling overflow: try options and conditional hiding](/en-US/docs/Web/CSS/CSS_anchor_positioning/Try_options_hiding)

> - [CSS anchor positioning](/en-US/docs/Web/CSS/CSS_anchor_positioning) module

Author: please add related features on pages as appropriate.
Inlcude "module" after each module link in see alsos
add the handling overflow as appropriate (not 100% sure it's needed here, but doesn't hurt). Its not in many pages since it was created after the PRs were submitted

Reviewer: I'll update the see also links as appropriate on each feature page.
